### ISSUE TITLE
Feature: yank url from bookmarks/history

### ISSUE THREAD

**Issue by kwstannard** [tridactyl/tridactyl#1886]
Tridactyl version: 1.17.0pre3184-691b830
Firefox version: Mozilla Firefox 70.0
Operating system: mac

I am in a role where I am finding myself wanting to copy the same urls regularly and currently I need to visit the URL first and then `yy`. Not a huge deal, but it would save a few seconds if I could just do something like this:

`:yank some words`
\<list of bmarks and history completions appears>
\<tab to url>
\<hit enter>

A cursory glance makes me think this would be possible by adding `yank` to the history completions list, but I thought I would put this issue up before attempting to do the change.

**Comment by bovine3dom** [comment]
This approach would odd but feels strange to me.

Perhaps an ex-mode command and bind to yank the highlighted completion? E.g. `<C-o>yy`?

**Comment by kwstannard** [comment]
That was a fast response, thank you.

I have never used ex-mode commands, but are you suggesting something like this?

`:open some words`
\<list of bmarks and history completions appears>
\<tab to url>
`<C-o>yy`
`<esc>`

**Comment by bovine3dom** [comment]
Yes, although you could have the `<Esc>` command as part of the previous bind.

It's very likely that you have used ex-mode commands - it's how the command line works. E.g. https://github.com/tridactyl/tridactyl/blob/a649b80b812908834a05084520852227de5a76c5/src/lib/commandline_cmds.ts#L108 and https://github.com/tridactyl/tridactyl/blob/a649b80b812908834a05084520852227de5a76c5/src/lib/config.ts#L95

**Comment by kwstannard** [comment]
Sorry, this is all new so I am probably suggesting things without fully understanding what my words mean. You want me write a new function that is kind of like accept_line that would be triggered by `<C-o>yy`?

I feel like it might be a little confusing to have both tabopen and open have an exit excmd that yanks the suggestion instead of doing an open or tabopen.

**Comment by bovine3dom** [comment]
It'd be more like insert_completion: https://github.com/tridactyl/tridactyl/blob/9cf7d9c74ef2642aeb1d2bd76bb746fd0d6c949c/src/lib/commandline_cmds.ts#L40

The command itself wouldn't exit, you'd just bind it to a composite of the two (if we wanted that).

**Comment by bovine3dom** [comment]
The `<C-o>` prefix is inspired by Vim, btw - it lets you run a normal mode command while you're in insert mode, so I hope it wouldn't be too surprising to users.

**Comment by kwstannard** [comment]
I didn't know about C-o in vim and I have been using it for 7 years. VIM is full of surprises unfortunately. 😄 

I might let this sit for a bit since I don't know if I have time to write out a whole new function in an unfamiliar repository in an unfamiliar language and I may be the only person who wants this anyways. It might be better for everyone if I just run my own patch with my `:yank` suggestion.

**Comment by bovine3dom** [comment]
I'm happy to merge a patch that adds `yank` completions, I just thought this would be cooler :)

**Comment by kwstannard** [comment]
Okay, there is also no reason we couldn't have both :) 

I will hopefully have some time this weekend.

**Comment by bovine3dom** [comment]
Fixed in [64174e0](https://github.com/tridactyl/tridactyl/commit/64174e0b03171a2899000cc4f62a9374ee86c55e) six years ago O_O

**Comment by kwstannard** [comment]
Thanks!
